Api/Validator.php

```diff
@@ -131,9 +131,6 @@ public function validateCategories($categories = []) {
         }
         
         public function validateProduct($product) {
-            foreach ($product['categories'] as $category) {
```

Comment:
Why have you removed this?

Reply:
Found out there is validation for categories for each product that was set to required. After check in the documentation of the API i found no required status for category id's for both product and productBatch and removed since it's not needed and was breaking the sync of deleted products since they have no data about categories.

Reply:
But do you validate the deleted products with the `validateProduct` method?

Reply:
Yes, client->productBatch($serializedProducts) is executing validation for productBatch before sync.

Reply:
This is validating the type of each object in the categories array. I think it's preferable to stay so that we don't send invalid data (think null, empty strings, etc).
Why would this bother the validation of deleted products? If the `categories` property is an empty array then this would run  fine?

Reply:
Yes you are right, it would run fine, I just saw that this categories property is not required in the api docs and that currently we are validating only the required fields, so i thought it's something we missed up.